packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -133,10 +134,24 @@ export default function createCssVarsProvider(options) {
       colorScheme = ctx.colorScheme;
     }
 
-    const memoTheme = React.useMemo(() => {
-      // `colorScheme` is undefined on the server and hydration phase
-      const calculatedColorScheme = colorScheme || restThemeProp.defaultColorScheme;
+    if (process.env.NODE_ENV !== 'production') {
+      if (forceRecalculateTheme && !restThemeProp.vars) {
+        console.warn(
+          [
+            'MUI: The `forceRecalculateTheme` prop should only be used with CSS theme variables.',
+            'This prop should be used for a specific use case and is not recommended for general usage.',
```

Comment:
```suggestion
            'Note that it will slow down the app when changing between modes, so use it only when there is no possible solution left',
            'If you are not sure about your use case, feel free to create a [new issue](https://github.com/mui/material-ui/issues/new) to let us help',
```